### PR TITLE
Md/gh ratelimit fixes

### PR DESCRIPTION
**Pull request by Ardiea**
### What are the relevant tickets?
https://github.com/mitodl/ol-concourse-github-issues/issues/62

### Description (What does it do?)
 Updated the gh issues resource defaults to use app auth rather than token. Added app auth config to vault mount points for concourse.

### CODE REVIEW

File: src/ol_concourse/lib/resources.py [hunk 1]
```diff
@@ -91,14 +91,18 @@ def github_issues(  # noqa: PLR0913
     name: Identifier,
     repository: str,
     issue_prefix: str,
+    auth_method: Literal["token", "app"] = "token",
     access_token: str = "((github.issues_resource_access_token))",  # noqa: S107
+    app_id: str = "((github.issues_resource_app_id))",
+    app_installation_id: str = "((github.issues_resource_app_installation_id))",
+    private_ssh_key: str = "((github.issues_resource_private_ssh_key))",
```
Comment (blarghmatey) [inline src/ol_concourse/lib/resources.py:98]: Where is this value coming from?

File: src/ol_concourse/lib/resources.py [hunk 2]
```diff
@@ -91,14 +91,18 @@ def github_issues(  # noqa: PLR0913
     name: Identifier,
     repository: str,
     issue_prefix: str,
+    auth_method: Literal["token", "app"] = "token",
```
Comment (blarghmatey) [inline src/ol_concourse/lib/resources.py:94]: Should we just default this to "app" rather than having to override it explicitly in other places?

**Review by feoh — APPROVED** [review]

File: src/ol_concourse/lib/resources.py (hunk 1 above)
Comment (Ardiea) [reply src/ol_concourse/lib/resources.py:98]: Like where did the key come from? Github generates it downloads it when you setup an application. It is being stored into vault else where in this PR.

File: src/ol_concourse/lib/resources.py (hunk 2 above)
Comment (Ardiea) [reply src/ol_concourse/lib/resources.py:94]: Maybe? I was operating on the "don't break what is currently running" mindset but I'm not sure that applies here.

**Pull request merged**